feature-libs/cart/import-export/components/import-to-cart/import-to-cart.service.ts

```diff
@@ -71,4 +82,62 @@ export class ImportToCartService {
     const patternRegex = new RegExp(/(?<=\s|^)\d+(?=\s|$)/);
     return data.every((row) => patternRegex.test(row[1]));
   }
+
+  /**
+   * Returns observable which emits on every added product success or failure.
+   *
+   * It completes when all entries are added.
+   *
+   * HOW TO USE IT IN COMPONENT:
+   * We need a logic for a summary (i.e. convert partial success to a failure)
+   */
+  protected getResults(cartId: string): Observable<ProductImportInfo> {
+    return this.actionsSubject.pipe(
+      ofType(
+        CartActions.CART_ADD_ENTRY_SUCCESS,
+        CartActions.CART_ADD_ENTRY_FAIL
+      ), // TODO: other product types (i.e. configurable product) might use different actions for addding to cart
+      filter(
+        (
+          action: CartActions.CartAddEntrySuccess | CartActions.CartAddEntryFail
+        ) => action.payload.cartId === cartId
+      ),
+      map((action) => this.mapMessages(action))
+    );
+    // TODO: don't retrurn RAW NGRX ACTIONS
+    // ALTERNATIVE IDEA:
+    // add each entry one by one, then on success/failure emit a result
+    // and then add next entry
+  }
+
+  protected mapMessages(
+    action: CartActions.CartAddEntrySuccess | CartActions.CartAddEntryFail
+  ): ProductImportInfo {
+    const { productCode } = action.payload;
+    if (action instanceof CartActions.CartAddEntrySuccess) {
+      const { quantity, quantityAdded, entry, statusCode } = action.payload;
+      if (statusCode === ProductImportStatus.LOW_STOCK) {
+        return {
+          productCode,
+          statusCode,
+          productName: entry?.product?.name,
+          quantity,
+          quantityAdded,
+        };
+      }
+      if (statusCode === ProductImportStatus.SUCCESS) {
+        return { productCode, statusCode };
+      }
+    } else if (action instanceof CartActions.CartAddEntryFail) {
+      const { error } = action.payload;
+      if (error?.details[0]?.type === 'UnknownIdentifierError') {
+        return {
+          productCode,
+          statusCode: ProductImportStatus.UNKNOWN_IDENTIFIER,
+        };
+      }
+    }
+    console.error('Unrecognized cart add entry action type', action);
```

Comment:
👀

Reply:
the return has been added for typescript strict restriction, console error for easier pointing some unhandled action types, I hope it could be helpful

Reply:
Thanks for explanation. 

We should not log anything to console in production, unless really justified. 

If you really want for dev convenience to log something, we should wrap it in a condition if(isDevMode()). 

Btw. What about relaxing the severity from .error to console.warn?

When writing a message please balance the following rules:
- it will increase the JS bundle size - so don’t add it at all if not needed or at least keep it short
- for customers it will be logged from vendor.js without a context, so be clear what exactly went wrong